test/blackbox-tests/test-cases/pkg/ocamlformat/binary-avoid-conflict-with-project.t

```diff
@@ -0,0 +1,110 @@
+Make sure that the "dune-project" does not use the OCamlFormat binary dev-tool.
+
+  $ . ../helpers.sh
+  $ mkrepo
+
+Make a fake ocamlformat:
+  $ mkdir ocamlformat
+  $ cd ocamlformat
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name ocamlformat))
+  > EOF
+  $ cat > ocamlformat.ml <<EOF
+  > let version = "0.26.2"
+  > ;;
+  > let () = print_endline ("formatted with version "^version)
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name ocamlformat))
+  > EOF
+  $ cd ..
+  $ tar -czf ocamlformat-0.26.2.tar.gz ocamlformat
+  $ rm -rf ocamlformat
+
+  $ mkpkg ocamlformat 0.26.2 <<EOF
+  > build: [
```

Comment:
It might be simpler if the fake ocamlformat was just a shell script rather than an ocaml program, though I don't think it's worth it to change it in this PR as it's already working.